webpack.config.ts

```diff
@@ -0,0 +1,356 @@
+  EEG_VIS_ENABLED: EEGVisEnabled,
+}));
+
+if (EEGVisEnabled !== 'true' && EEGVisEnabled !== '1' ) {
+  plugins.push(
+    new IgnorePlugin({
+      resourceRegExp: /react-series-data-viewer/,
+    })
+  );
+}
+
+/**
+ * Get the webpack entries of a given module, which is described by its name
+ * and its entry points.
+ *
+ * @returns A list of two-element tuples mapping each entry name (exemple
+ * 'login/loginIndex') to its webpack entry.
+ */
+function makeModuleEntries(moduleName: string, files: string[]) {
+  // Check if a project override exists for the module.
+  const basePath = fs.existsSync(`./project/modules/${moduleName}`)
+    ? `./project/modules/${moduleName}/`
+    : `./modules/${moduleName}/`;
+
+  return files.map((fileName) => ([moduleName + '/' + fileName, {
+    import: basePath + 'jsx/' + fileName,
+    filename: basePath + 'js/' + fileName + '.js',
+    library: {
+      name: ['lorisjs', moduleName, fileName],
+      type: 'window',
+    },
+  }]));
+}
+
+// Add entries for project overrides.
+if (fs.existsSync('./project/webpack-project.config.js')) {
+  const projectModules: { [x: string]: string[] }
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    = require('./project/webpack-project.config.js');
+
+  for (const [moduleName, files] of Object.entries(projectModules)) {
+    if (moduleName in lorisModules) {
+      lorisModules[moduleName].push(...files);
+    } else {
+      lorisModules[moduleName] = files;
+    }
+  }
+}
+
+// Only build the given target module if there is one in the current
+// environment.
+const target = process.env.target;
+if (target) {
+  if (target in lorisModules) {
+    lorisModules = {
+      [target]: lorisModules[target],
+    };
+  } else {
+    console.error(`Target module '${target}' not found.`);
+    process.exit(1);
+  }
+}
+
+// Transform the mapping of LORIS modules to a mapping of webpack entry points.
+const entries = Object.fromEntries(
+  Object.entries(lorisModules)
+    .map(([name, files]) => makeModuleEntries(name, files))
+    .flat()
+);
+
+const configs: webpack.Configuration[] = [];
+
+configs.push({
+  entry: {
+    PaginationLinks: './jsx/PaginationLinks.js',
+    StaticDataTable: './jsx/StaticDataTable.js',
+    MultiSelectDropdown: './jsx/MultiSelectDropdown.js',
+    Breadcrumbs: './jsx/Breadcrumbs.js',
+    CSSGrid: './jsx/CSSGrid.js',
+    Help: './jsx/Help.js',
+    ...entries,
+  },
+  output: {
+    path: __dirname,
+    filename: './htdocs/js/components/[name].js',
+    library: ['lorisjs', '[name]'],
+    libraryTarget: 'window',
+  },
+  externals: {'react': 'React', 'react-dom': 'ReactDOM'},
+  devtool: 'source-map',
+  plugins,
+  optimization,
+  resolve,
+  module,
+  stats: 'errors-warnings',
+});
+
+// HACK: For some reason, the electrophysiology session view only compiles if
+// it uses a separate (although possibly identical) configuration.
+if (!target || target === 'electrophysiology_browser') {
```

Comment:
I'm curious why we are checking for !target as well here.

Reply:
`target` informs us of which single module we want to compile, but it can also be `undefined` if we want to compile all modules. We need to take into account both of these cases.